src/index.html

```diff
@@ -11,7 +11,146 @@
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    
+    <form
+      action="link" 
+      method="post"
+    >
+      <fieldset class="fieldset">
+        <legend>Personal information</legend>
+        <label class="input">
+          Surname:
+          <input 
+            type="text" 
+            name="Surname" 
+            autocomplete="off"
+          >
+        </label>
+        <label class="input">
+          Name:
+          <input           
+            type="text" 
+            name="Name" 
+            autocomplete="off"
+          >
+        </label>
+        <label class="input">
+          How old are You?
+          <input 
+            type="number" 
+            name="Age" 
+            min="1" 
+            max="100" 
+            value="12"
+          >
+        </label>
+        <label class="input">
+          Full date of birth:
+          <input 
+            type="date" 
+            name="Birth"
+          >
+        </label>
+        <label>
+          I accept the term of the agreement
+          <input 
+            type="checkbox" 
+            name="TermAccept" 
+            required
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="fieldset">
+        <legend>Registration</legend>
+        <label class="input">
+          E-mail:
+          <input 
+            type="email" 
+            name="email" 
+            placeholder="email@example.com"
+          >
+        </label>
+        <label>
+          Password:
+          <input 
+            type="password" 
+            name="password" 
+            minlength="5" 
+            maxlength="15"
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="fieldset">
+        <legend>An interesting fact about you!</legend>
+        <div class="input">
+          Do you love cats?
+          <input 
+            type="radio" 
+            name="cats"
+            id="yes"
+            value="y"
+          >
+          <label for="yes">Yes</label> 
+                        
+          <input
+            type="radio" 
+            name="cats" 
+            id="no"
+            value="n"
+          >
+          <label for="no">No</label>
+        </div>
+
+        <label class="input">
+          What is your favorite color:
+          <input type="color">
+        </label>
+
+        <label class="input">
+          What time do you go to bad?
+          <input 
+            type="time" 
+            name="TimeToBad"
```

Comment:
<img width="890" alt="Screenshot 2022-12-06 at 10 59 24" src="https://user-images.githubusercontent.com/104769168/205866462-5b3dfc52-27ce-4897-9f20-8ab889029626.png">
```suggestion
            name="timeToBad"
```
fix name attributes according to the checklist all over the page